packages/agoric-cli/src/commands/inter.js

```diff
@@ -0,0 +1,376 @@
+          ...opts,
+        });
+        outputExecuteOfferAction(offer, stdout);
+        stderr.write(
+          'Now use `agoric wallet send ...` to sign and broadcast the offer.\n',
+        );
+      },
+    );
+
+  const parsePercent = v => {
+    const p = Number(v);
+    if (!(p >= -100 && p <= 100)) {
+      throw new InvalidArgumentError('must be between -100 and 100');
+    }
+    return p / 100;
+  };
+
+  bidCmd
+    .command('by-discount')
+    .description(
+      `Print an offer to bid on collateral based on discount from oracle price.`,
+    )
+    .requiredOption(
+      '--discount [percent]',
+      'bid discount (0 to 100) or markup (0 to -100) %',
+      parsePercent,
+    )
+    .requiredOption('--giveCurrency [number]', 'Currency to give', Number)
+    .requiredOption('--wantCollateral [number]', 'bid price', Number)
+    .option('--collateralBrand [string]', 'Collateral brand key', 'IbcATOM')
+    .option('--offerId [number]', 'Offer id', String, `bid-${clock()}`)
+    .action(
+      /**
+       * @param {{
+       *   discount: number,
+       *   giveCurrency: number,  wantCollateral: number,
+       *   collateralBrand: string,
+       *   offerId: string,
+       * }} opts
+       */
+      async ({ collateralBrand, ...opts }) => {
+        const { agoricNames } = await rpcTools();
+        const offer = Offers.auction.Bid(agoricNames.brand, {
+          collateralBrandKey: collateralBrand,
+          ...opts,
+        });
+        outputExecuteOfferAction(offer, stdout);
+        stderr.write(
+          'Now use `agoric wallet send ...` to sign and broadcast the offer.\n',
+        );
+      },
+    );
+
+  const normalizeAddress = literalOrName =>
+    normalizeAddressWithOptions(literalOrName, interCmd.opts(), {
+      execFileSync,
+    });
+
+  bidCmd
+    .command('list')
+    .description(
+      `Show status of bid offers.
+
+For example:
+
+$ inter bid list --from my-acct
+{"id":"bid-1679677228803","price":"9 IST/IbcATOM","give":{"Currency":"50IST"},"want":"5IbcATOM"}
+{"id":"bid-1679677312341","discount":10,"give":{"Currency":"200IST"},"want":"1IbcATOM"}
+`,
+    )
+    .requiredOption(
+      '--from <address>',
+      'wallet address literal or name',
+      normalizeAddress,
+    )
+    .action(async opts => {
+      const { agoricNames, vstorage, fromBoard } = await rpcTools();
+      const m = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+
+      const history = await vstorage.readFully(`published.wallet.${opts.from}`);
+
+      /** @type {{ Invitation: Brand<'set'> }} */
+      // @ts-expect-error XXX how to narrow AssetKind to set?
+      const { Invitation } = agoricNames.brand;
+      const coalescer = makeWalletStateCoalescer(Invitation);
+      // update with oldest first
+      for (const txt of history.reverse()) {
+        const { body, slots } = JSON.parse(txt);
+        const record = m.unserialize({ body, slots });
+        coalescer.update(record);
```

Comment:
consider a feature like,
```suggestion
        coalescer.updateSerialized(txt);
```

Reply:
making the coalescer aware of marshaling looks kinda awkward